src/index.ts

```diff
@@ -76,61 +44,79 @@ const addKeyValuePairToObject = (obj: any, keyValuePair: string): any => {
 /**
  * Parses one openattestation DNS-TXT record and turns it into an OpenAttestationsDNSTextRecord object
  * @param record e.g: '"openatts net=ethereum netId=3 addr=0x0c9d5E6C766030cc6f0f49951D275Ad0701F81EC"'
  */
-export const parseOpenAttestationRecord = (record: string): OpenAttestationDNSTextRecord => {
+export const parseOpenAttestationRecord = (record: string): GenericObject => {
   trace(`Parsing record: ${record}`);
   const keyValuePairs = record.trim().split(" "); // tokenize into key=value elements
-  const recordObject = {} as OpenAttestationDNSTextRecord;
+  const recordObject = {} as GenericObject;
   // @ts-ignore: we already checked for this token
   recordObject.type = keyValuePairs.shift();
-  keyValuePairs.reduce<OpenAttestationDNSTextRecord>(addKeyValuePairToObject, recordObject);
+  keyValuePairs.reduce<GenericObject>(addKeyValuePairToObject, recordObject);
   return recordObject;
 };
 
 /**
  * Currying function that applies a given dnssec result
  */
-const applyDnssecResults = (dnssecStatus: boolean) => (
-  record: OpenAttestationDNSTextRecord
-): OpenAttestationDNSTextRecord => {
+const applyDnssecResults = <T>(dnssecStatus: boolean) => (record: T): T => {
   return { ...record, dnssec: dnssecStatus };
 };
 
-/**
- * Returns true if the given object passes runtype validation for OpenAttestationDNSTextRecord
- * Turn on debug log to see exact validation failure messages if necessary
- * @param record An object that may conform to the OpenAttestationDNSTextRecord shape
- */
-export const isWellFormedOpenAttestationRecord = (record: OpenAttestationDNSTextRecord) => {
-  try {
-    OpenAttestationDNSTextRecord.check(record);
-    return true;
-  } catch (e) {
-    error(e);
-    return false;
-  }
+const parseOpenAttestationRecords = (recordSet: IDNSRecord[] = []): GenericObject[] => {
+  trace(`Parsing DNS results: ${JSON.stringify(recordSet)}`);
+  return recordSet
+    .map((record) => record.data)
+    .map((record) => record.slice(1, -1)) // removing leading and trailing quotes
+    .filter(isOpenAttestationRecord)
+    .map(parseOpenAttestationRecord);
 };
 
 /**
  * Takes a DNS-TXT Record set and returns openattestation document store records if any
  * @param recordSet Refer to tests for examples
  */
-export const parseDnsResults = (recordSet: IDNSRecord[] = []): OpenAttestationDNSTextRecord[] => {
-  trace(`Parsing DNS results: ${JSON.stringify(recordSet)}`);
-  return recordSet
-    .map((record) => record.data)
-    .map((record) => record.slice(1, -1)) // removing leading and trailing quotes
-    .filter(isOpenAttestationRecord)
-    .map(parseOpenAttestationRecord)
-    .filter(isWellFormedOpenAttestationRecord);
+export const parseDocumentStoreResults = (
+  recordSet: IDNSRecord[] = [],
+  dnssec: boolean
+): OpenAttestationDNSTextRecord[] => {
+  return parseOpenAttestationRecords(recordSet)
+    .reduce((prev, curr) => {
+      return OpenAttestationDNSTextRecordT.guard(curr) ? [...prev, curr] : prev;
```

Comment:
would .filter here be easier to read?

Reply:
it is doing both map and filter.
cannot do 1-1 mapping since not all generic records are of this type, so it filter out those are are not, and provide a typeguard for the filtered result, converting from the type `GenericObject` to `OpenAttestationDNSTextRecord`.

Reply:
right ok i missed that it's an array